private/buf/bufcli/errors.go

```diff
@@ -164,21 +166,42 @@ func NewTemplateNotFoundError(owner string, name string) error {
 // Note that this function will wrap the error so that the underlying error
 // can be recovered via 'errors.Is'.
 func wrapError(err error) error {
-	if err == nil || (err.Error() == "" && !rpc.IsError(err)) {
-		// If the error is nil or empty and not an rpc error, we return it as-is.
-		// This is especially relevant for commands like lint and breaking.
+	// If error is nil or empty and not a Connect error, we return it as-is.
+	// This is especially relevant for commands like lint and breaking.
+	if err == nil || (err.Error() == "" && !isConnectError(err)) {
 		return err
 	}
-	rpcCode := rpc.GetErrorCode(err)
-	switch {
-	case rpcCode == rpc.ErrorCodeUnauthenticated, isEmptyUnknownError(err):
-		return errors.New(`Failure: you are not authenticated. Create a new entry in your netrc, using a Buf API Key as the password. For details, visit https://docs.buf.build/bsr/authentication`)
-	case rpcCode == rpc.ErrorCodeUnavailable:
-		return fmt.Errorf(`Failure: the server hosted at that remote is unavailable: %w`, err)
+	// Attempt to convert err to a Connect error.  If it can be converted, then interpret it and return an intuitive error
+	if connectError := (&connect.Error{}); errors.As(err, &connectError) {
+		connectCode := connectError.Code()
+		connectMessage := connectError.Message()
+		switch {
+		case connectCode == connect.CodeUnauthenticated, isEmptyUnknownError(err):
+			return errors.New(`Failure: you are not authenticated. Create a new entry in your netrc, using a Buf API Key as the password. For details, visit https://docs.buf.build/bsr/authentication`)
+		case connectCode == connect.CodeUnavailable:
+			msg := `Failure: the server hosted at that remote is unavailable.`
+
+			// If the returned error is Unavailable, then determine if this is a DNS error.  If so, get the address used
+			// so that we can display a more helpful error message.
+			if dnsError := (&net.DNSError{}); errors.As(err, &dnsError) && dnsError.IsNotFound {
+				return fmt.Errorf(`%s Are you sure "%s" is a valid remote address?`, msg, buftransport.TrimAPISubdomain(dnsError.Name))
+			}
+
+			return fmt.Errorf(msg)
+		}
+		return fmt.Errorf("Failure: %s", connectMessage)
 	}
+
+	// Error was not empty, but was not a Connect error
```

Comment:
I don't think that would happen.  Because the first if statement will return if the error is empty, but not a Connect error

```
if err == nil || (err.Error() == "" && !isConnectError(err)) {
```

If it makes it past that and the error is empty, we know it must be a Connect error then, which means it will enter the second if statement:

```
if connectError := (&connect.Error{}); errors.As(err, &connectError) {
```

and return from there.